posthog/batch_exports/service.py

```diff
@@ -847,3 +849,46 @@ async def afetch_batch_export_runs_in_range(
     ).order_by("data_interval_start")
 
     return [run async for run in queryset]
+
+
+def fetch_earliest_backfill_start_at(
+    *,
+    team_id: int,
+    model: str,
+    interval_time_delta: dt.timedelta,
+    exclude_events: collections.abc.Iterable[str] | None = None,
+    include_events: collections.abc.Iterable[str] | None = None,
+) -> dt.datetime:
+    """Get the earliest start_at for a batch export backfill.
+
+    Note that this query will return the unix epoch if no events or persons exist.
+    """
+    interval_seconds = interval_time_delta.total_seconds()
```

Comment:
Not sure how relevant this is, but `total_seconds` returns a `float`.

Reply:
seems like ClickHouse doesn't mind it being a float but maybe makes sense to convert to an int for readability